src/main/java/com/amazonaws/services/dynamodbv2/datamodeling/AttributeEncryptor.java

```diff
@@ -58,9 +64,26 @@ public DynamoDBEncryptor getEncryptor() {
     public Map<String, AttributeValue> transform(final Parameters<?> parameters) {
         // one map of attributeFlags per model class
         final ModelClassMetadata metadata = getModelClassMetadata(parameters);
+
+        final Map<String, AttributeValue> attributeValues = parameters.getAttributeValues();
+        // If this class is marked as "DoNotTouch" then we know our encryptor will not change it at all
+        // so we may as well fast-return and do nothing. This also avoids emitting errors when they would not apply.
+        if (metadata.doNotTouch) {
+            return attributeValues;
+        }
+
+        // When AttributeEncryptor is used without SaveBehavior.CLOBBER, it is trying to transform only a subset
+        // of the actual fields stored in DynamoDB. This means that the generated signature will not cover any
+        // unmodified fields. Thus, upon untransform, the signature verification will fail as it won't cover all
+        // expected fields.
+        if (parameters.isPartialUpdate()) {
+            LOG.error("Use of AttributeEncryptor without SaveBehavior.CLOBBER is an error and can result in data-corruption. " +
```

Comment:
For anyone who is not using CLOBBER today, there will be tons of errors in the log and the operation is not actually failing.

Reply:
That is correct because they are in a dangerous error case and are at risk for data-corruption. Minor software changes outside of their (or our) control could result in this going from a "working due to luck" state to a "broken" state. We consider it sufficiently important to notify people of this dangerous configuration that an ERROR level log is appropriate.

The next minor (non-patch) version will throw an exception in this case. Please see #32